packages/shaku-code-annotate-sugar-high/src/index.mts

```diff
@@ -0,0 +1,395 @@
+export { defaultCode } from "./defaultCode.mjs";
+import {
+  ShakuDirectiveHighlightInline,
+  parseLine,
+  renderComponent,
+} from "shaku-code-annotate-core";
+import * as sh from "sugar-high";
+import {
+  renderSourceLine,
+  renderSourceLineWithInlineHighlight,
+} from "./render.mjs";
+import { Token } from "./types.mjs";
+
+// @ts-ignore
+// this type is missing from sugar-high
+const TokenTypes = sh.SugarHigh.TokenTypes;
+
+type RawToken = [number, string];
+
+export function highlight(code: string) {
+  const tokens = sh.tokenize(code);
+  const lines = toLines(tokens);
+  const parsedLines = parseLines(lines, "javascript", true);
+  console.log(parsedLines);
+
+  let html = `<pre class="shaku"><code>`;
+
+  const hasFocus = hasShakuDirectiveFocus(parsedLines);
+
+  let shouldHighlighNextSourceLine = false;
+  let shouldDimNextSourceLine = false;
+  let shouldFocusNextSourceLine = false;
+  let isHighlightingBlock = false;
+  let isDimBlock = false;
+  let isFocusBlock = false;
+  let shakuDirectiveHighlightInline: null | {
+    type: "shaku";
+    line: ShakuDirectiveHighlightInline;
+    offset: number;
+  } = null;
+
+  for (let i = 0; i < parsedLines.length; i++) {
+    const line = parsedLines[i];
+
+    const isShakuLine = line.type === "shaku" && !line.line.config.isEscaped;
+    if (isShakuLine) {
+      const shakuLine = line.line;
+      switch (shakuLine.type) {
+        case "DirectiveCallout": {
+          const arrowOffset = shakuLine.config.offset;
+          const directiveOffset = arrowOffset + line.offset;
+          let minOffset = directiveOffset;
+          const contents = [];
+
+          let j = i + 1;
+          while (j < parsedLines.length) {
+            const nextLine = parsedLines[j];
+            if (
+              nextLine.type !== "shaku" ||
+              nextLine.line.type !== "AnnotationLine"
+            ) {
+              break;
+            }
+
+            minOffset = Math.min(
+              minOffset,
+              nextLine.line.config.offset + nextLine.offset
+            );
+            contents.push(
+              // in sugar-high, we don't need md parsing
+              nextLine.line.config.content
+            );
+            j += 1;
+          }
+          html += renderComponent({
```

Comment:
- [ ] separate renderComponent() into two, one supports HTML and sanitize. one does escape.